File: app/src/main/java/com/testosterolapp/unrd/serverCommunication/AppController.java
package com.testosterolapp.unrd.serverCommunication;

import android.content.Context;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

public class AppController {

    private static AppController AppControllerSingleton;
    private RequestQueue mRequestQueue;
    private static Context mContext;

    private AppController(Context context) {
        mContext = context;
        mRequestQueue = getRequestQueue();

    }

    public static synchronized AppController getInstance(Context context) {
        if (AppControllerSingleton == null) {
            AppControllerSingleton = new AppController(context);
        }
        return AppControllerSingleton;
    }

    public RequestQueue getRequestQueue() {
        if (mRequestQueue == null) {
            // getApplicationContext() is key, it keeps you from leaking the
            // Activity or BroadcastReceiver if someone passes one in.
            mRequestQueue = Volley.newRequestQueue(mContext.getApplicationContext());
        }
        return mRequestQueue;
    }

    public <T> void addToRequestQueue(Request<T> req) {
        getRequestQueue().add(req);
    }

    public void cancelPendingRequests(Object tag) {
        if (mRequestQueue != null) {
            mRequestQueue.cancelAll(tag);
        }
    }
}

File: app/src/main/java/com/testosterolapp/unrd/serverCommunication/AppController.kt
package com.testosterolapp.unrd.serverCommunication

import android.content.Context
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley

class AppController private constructor(private var mContext: Context) {
    private var mRequestQueue: RequestQueue?

    // getApplicationContext() is key, it keeps you from leaking the
    // Activity or BroadcastReceiver if someone passes one in.
    val requestQueue: RequestQueue?
        get() {
            if (mRequestQueue == null) {
                // getApplicationContext() is key, it keeps you from leaking the
                // Activity or BroadcastReceiver if someone passes one in.
                mRequestQueue = Volley.newRequestQueue(this.mContext.applicationContext)
            }
            return mRequestQueue
        }

    fun <T> addToRequestQueue(req: Request<T>?) {
        requestQueue!!.add(req)
    }

    fun cancelPendingRequests(tag: Any?) {
        if (mRequestQueue != null) {
            mRequestQueue!!.cancelAll(tag)
        }
    }

    companion object {
        private var AppControllerSingleton: AppController? = null
        @Synchronized
        fun getInstance(context: Context): AppController? {
            if (AppControllerSingleton == null) {
                AppControllerSingleton = AppController(context)
            }
            return AppControllerSingleton
        }
    }

    init {
        mRequestQueue = requestQueue
    }
}